### ISSUE TITLE
Generated dhcpd.conf lacks Dynamic DNS information

### ISSUE THREAD

**Issue by rmx-it** [opnsense/core#1023]
Hi,
I'm using opnsense 16.1.17.
I have three LAN-Segments with two of them having dynamic DNS entries.

For the LAN segment the settings look like:
![image](https://cloud.githubusercontent.com/assets/11737548/16178697/ef41c320-364f-11e6-9151-69db1c59db45.png)

For the WLAN segment similarly:
![image](https://cloud.githubusercontent.com/assets/11737548/16178704/2eeedbfc-3650-11e6-84e6-5518eab592bf.png)

The problem is:
 The dynamic DNS settings don't get added in the created dhcpd.conf at all.

Any clues?

Thannks in advance!

Greetings
---
Mike

**Comment by AdSchellevis** [comment]
@rmx-it can you test https://github.com/opnsense/core/commit/5fa09cd4d86052aaa8ff9840499c26721e7fa18c ? If this works, there seem to be quite some duplicate / non functional code which should be removed as well.

The "Dynamic DNS" settings from the server should contain the registration settings in that case, the static registration should only contain the dnsdomain then.

To test, use this on a console:

```
opnsense-patch  5fa09cd
```

**Comment by rmx-it** [comment]
Hi,
still the key info is missing:
For the LAN segment (after the patch):
````
...
subnet 10.0.0.0 netmask 255.255.255.0 {
  pool {
    option domain-name-servers 10.0.0.2;
    ddns-update-style interim;
    range 10.0.0.100 10.0.0.200;
  }

  option routers 10.0.0.2;
  option domain-name "rm-i.net";
  option domain-search "rm-i.net","wlan.rm-i.net";
  ddns-domainname "rm-i.net";
  option domain-name-servers 10.0.0.2;
  default-lease-time 3600;
}
...
````
and below the file:
````
...
ddns-update-style interim;
update-static-leases on;
zone rm-i.net. {
  primary 127.0.0.1;
}
zone 0.0.10.in-addr.arpa {
  primary 127.0.0.1;
}
...
````

Greetings
---
Mike

**Comment by fichtner** [comment]
@rmx-it I'm working on this, will have something to test tonight

One thing I saw in the code was that reusing the same name/key may be a problem, but that's for after the next fix to try/clean up

**Comment by fichtner** [comment]
How's this?

    # opnsense-patch ae6b495

**Comment by rmx-it** [comment]
Hi,
looks better.
The key get exported, the zone information too.

The remaining problem:
If I use the same key for multiple interfaces, the same key gets exported multiple times...

Workaround: Use different keys for each segment.
Solution: Parse for duplicates on export or make the keys a select-option table.

Thanks anyway for your fast efforts!

cu
...
Mike

**Comment by fichtner** [comment]
Ok, exactly what I expected. It can be fixed, although the model is a bit weak in that regard. The last or first key entered for the same domain will have to "win" so a typo may be disruptive. Don't want to change the way the GUI or data model works at this point almost having reached 16.7 RC. Patch follows in a bit...

Thanks for testing so far :)

**Comment by fichtner** [comment]
Hey Mike,

Here you go!

    # opnsense-patch 4dde1f0


Cheers,
Franco

**Comment by rmx-it** [comment]
Hi,
nearly perfect, but not fully there:

If using the same key in two segments, the key gets exported only once.
Thats good.

But only the first zone gets the key statement, the second not.
Thats bad :-)

If using different keys for the two segments, everything is fine!

Cheers,
Mike

**Comment by fichtner** [comment]
Hey Mike,

Third time's the charm! :)

    # opnsense-patch e287db60


Cheers,
Franco

**Comment by rmx-it** [comment]
Hi,
yes, now it looks right.

Thanks!

Greetings,
Mike

**Comment by fichtner** [comment]
Yay, thanks for testing. Will be shipped in 16.1.18 early next week.
